CMakeLists.txt

```diff
@@ -1,11 +1,22 @@
-project(HighFive)
 cmake_minimum_required(VERSION 3.0)
 
-enable_testing()
+if(POLICY CMP0048)
+  cmake_policy(SET CMP0048 NEW)  # package version behavior added in cmake 3.0
+endif()
+if(POLICY CMP0054)
+  cmake_policy(SET CMP0054 NEW)  # Only interpret if() arguments as variables or keywords when unquoted
+endif()
+if(POLICY CMP0074)
+  cmake_policy(SET CMP0074 NEW)  # find_package behavior added in cmake 3.12
+endif()
 
 set(HIGHFIVE_VERSION_MAJOR 2)
 set(HIGHFIVE_VERSION_MINOR 0)
 
+project(HighFive VERSION "${HIGHFIVE_VERSION_MAJOR}.${HIGHFIVE_VERSION_MINOR}")
```

Comment:
Setting the version automatically creates these variables for you (at least, `HighFive_VERSION_MAJOR`, etc, that's the point). If you rename the project to HIGHFIVE here, then `HIGHFIVE_VERSION_MAJOR` will be set.